R/convert.R

```diff
@@ -16,9 +16,13 @@
 #'   to work because those methods will return `classParent` objects, not
 #'   `classChild` objects.
 #'
-#' `convert()` provides a default implementation when `from` inherits from
-#' `to`. This default strips any properties that `from` possesses that `to`
-#' does not.
+#' `convert()` provides two default implementations:
+#'
+#' 1. When `from` inherits from `to`, it strips any properties that `from`
+#'    possesses that `to` does not (downcasting).
+#' 2. When `to` is a subclass of `from`'s class, it creates a new object of
```

Comment:
I think this case is already covered by the `if (class_inherits(from, to))` branch. We could add an early `if` branch like this:

```
  } else if (identical(dispatch[[1L]], dispatch[[2L]])) {
    # `from` is already a `to`
    from
  }
```

But it would only help with readability and maybe be a little faster.

Reply:
Oh yeah, makes sense.